File: 프로그래머스/2/132265. 롤케이크 자르기/롤케이크 자르기.py
def solution(topping):
    answer = 0
    
    lt = set()
    rt = {}
    
    for t in topping:
        if t in rt:
            rt[t] +=1
        else:
            rt[t] = 1
            
    for t in topping:
        lt.add(t)
        rt[t] -=1
        if rt[t] == 0:
            del rt[t]
        if len(lt) == len(rt):
            answer+=1
    
    
    return answer

File: 프로그래머스/2/132265. 롤케이크 자르기/롤케이크 자르기.java
import java.util.*;
class Solution {
    public int solution(int[] topping) {
        int answer = 0;
        int n = topping.length;
        Map<Integer,Integer> map = new HashMap<>();
        Set<Integer> lt = new HashSet<>();
        for(int i=0; i<n;i++){
            map.put(topping[i], map.getOrDefault(topping[i],0)+1);
        }
        
        for(int i=0;i<n;i++){
            int t = topping[i];
            
            lt.add(t);
            map.put(t, map.get(t)-1);
            if(map.get(t)==0){
                map.remove(t);
            }
            if(map.size() == lt.size()) answer++;
        }
        
        return answer;
    }
}